### PR TITLE
Add blocking counter

### LINKED ISSUE #6999: Implement blocking counter which would be used with the global thread pool

### PR DESCRIPTION
**Pull request by Yancey1989**
Fixed #6999

### CODE REVIEW

File: paddle/framework/blocking_counter.h [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  ~BlockingCounter() {}
+
+  /**
+   * @breif   Wait untile the count was decreased to 0
+   */
+  void Wait() {
+    if (cnt_ == 0) return;
+    std::unique_lock<std::mutex> lock(done_m);
+    done_cv.wait(lock, [=] { return done_ == true; });
```
Comment (typhoonzero) [inline paddle/framework/blocking_counter.h]: Just wait for `cnt_ == 0` condition, then `done_` is not needed.

File: paddle/framework/blocking_counter.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  /**
+   * @breif   Descrease the count by 1
+   */
+  void DecreaseCount() {
+    cnt_.fetch_sub(1);
+    if (cnt_ != 0) return;
```
Comment (typhoonzero) [inline paddle/framework/blocking_counter.h]: Use `if (cnt_-- !=0)` is simpler.

Comment (Yancey1989) [reply paddle/framework/blocking_counter.h]: Thanks, simpler is great, and I think `if (cnt_-- != 1)` is correct.

Comment (Yancey1989) [reply paddle/framework/blocking_counter.h]: Done.

File: paddle/framework/blocking_counter.h (hunk 1 above)
Comment (Yancey1989) [reply paddle/framework/blocking_counter.h]: Done.

**Comment by Yancey1989** [comment]
Doesn't need this feature, close this PR.

**Pull request closed without merging**